<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.251">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Afshan's Portfolio – overview</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">(Untitled)</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Afshan’s Portfolio</a> 
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">Misannotated lncRNAs</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../misannotated_lncrnas/Overview.html" class="sidebar-item-text sidebar-link active">Identifying misannotated lncRNAs</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">fastai</span>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">Notes</span>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#identifying-misannotated-lncrnas" id="toc-identifying-misannotated-lncrnas" class="nav-link active" data-scroll-target="#identifying-misannotated-lncrnas">Identifying misannotated lncRNAs</a>
  <ul class="collapse">
  <li><a href="#alright-whats-the-problem-were-looking-at" id="toc-alright-whats-the-problem-were-looking-at" class="nav-link" data-scroll-target="#alright-whats-the-problem-were-looking-at">Alright! What’s the problem we’re looking at?</a></li>
  <li><a href="#what-are-the-issues-with-identifying-these-misannotated-lncrnas" id="toc-what-are-the-issues-with-identifying-these-misannotated-lncrnas" class="nav-link" data-scroll-target="#what-are-the-issues-with-identifying-these-misannotated-lncrnas">What are the issues with identifying these <code>misannotated lncRNAs</code>?</a></li>
  <li><a href="#so-what-are-training-dynamics" id="toc-so-what-are-training-dynamics" class="nav-link" data-scroll-target="#so-what-are-training-dynamics">So, what are training dynamics?</a></li>
  <li><a href="#uh-does-this-training-dynamics-stuff-really-work-when-a-large-number-of-sequences-might-be-mislabelled" id="toc-uh-does-this-training-dynamics-stuff-really-work-when-a-large-number-of-sequences-might-be-mislabelled" class="nav-link" data-scroll-target="#uh-does-this-training-dynamics-stuff-really-work-when-a-large-number-of-sequences-might-be-mislabelled">Uh, does this training dynamics stuff really work when a large number of sequences might be mislabelled?</a></li>
  <li><a href="#ok-so-this-computational-method-works.-but-how-well-do-the-results-agree-with-exisiting-experimental-results" id="toc-ok-so-this-computational-method-works.-but-how-well-do-the-results-agree-with-exisiting-experimental-results" class="nav-link" data-scroll-target="#ok-so-this-computational-method-works.-but-how-well-do-the-results-agree-with-exisiting-experimental-results">Ok, so this computational method works. But how well do the results agree with exisiting experimental results?</a></li>
  <li><a href="#hmm-this-is-useful-whats-the-best-way-to-use-this-data" id="toc-hmm-this-is-useful-whats-the-best-way-to-use-this-data" class="nav-link" data-scroll-target="#hmm-this-is-useful-whats-the-best-way-to-use-this-data">Hmm, this is useful! What’s the best way to use this data?</a></li>
  <li><a href="#and-thats-a-wrap" id="toc-and-thats-a-wrap" class="nav-link" data-scroll-target="#and-thats-a-wrap">And that’s a wrap!</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">



<section id="identifying-misannotated-lncrnas" class="level1">
<h1>Identifying misannotated lncRNAs</h1>
<p>The aim of this overview is to explain this work as simply as possible.</p>
<p>For more biological and technical details, check out the <a href="https://academic.oup.com/bioinformatics/article/39/1/btac821/6960922">paper in Bioinformatics</a>!</p>
<section id="alright-whats-the-problem-were-looking-at" class="level3">
<h3 class="anchored" data-anchor-id="alright-whats-the-problem-were-looking-at">Alright! What’s the problem we’re looking at?</h3>
<ul>
<li>Long non-coding RNAs (lncRNAs) are defined as non-coding (i.e.&nbsp;don’t code for proteins) RNAs greater than 200 nucleotides in length.</li>
<li>However, recently it was discovered that some open-reading frames within lncRNAs <em>can</em> be translated into small proteins (micropeptides).</li>
<li>Such lncRNAs are called <code>misannotated lncRNAs</code> and we want to identify them.</li>
</ul>
</section>
<section id="what-are-the-issues-with-identifying-these-misannotated-lncrnas" class="level3">
<h3 class="anchored" data-anchor-id="what-are-the-issues-with-identifying-these-misannotated-lncrnas">What are the issues with identifying these <code>misannotated lncRNAs</code>?</h3>
<ul>
<li>Experimental methods:
<ul>
<li>Expensive!</li>
<li>Time-consuming!</li>
<li>Cell specific. Need to do a very comprehensive search!</li>
<li>Differences across species?</li>
</ul></li>
<li>Computational methods:
<ul>
<li>Coding/Non-Coding classifiers: Normal prediction methods are hamstrung because we don’t know which lncRNAs are misannotated i.e.&nbsp;training datasets have wrong labels.</li>
</ul></li>
</ul>
<p>So there are 2 main problems: * A method able to identify <code>misannotated lncRNAs</code> by sequence alone would be amazing! * It would save time, experimental costs, while being independent of cell-type and species. * If we could generate a list of possible candidate <code>misannotated lncRNAs</code>, experimental validation would be a lot more directed. * Current labels of lncRNAs are wrong. Therefore, a prediction method that doesn’t rely completely on labels would be ideal.</p>
<p>Our method addresses these problems by relying on <em>training dynamics of deep learning methods</em> to solve the problem of identifying <code>misannotated lncRNAs</code>.</p>
</section>
<section id="so-what-are-training-dynamics" class="level3">
<h3 class="anchored" data-anchor-id="so-what-are-training-dynamics">So, what are training dynamics?</h3>
<ul>
<li>Relying on training dynamics of deep learning models means that:
<ul>
<li>At the end of every training epoch, we have the model make coding/non-coding predictions for the sequences in our test set. We save these predictions.</li>
<li>Finally, we take the mean and standard deviation of coding/non-coding predictions over all epochs.</li>
</ul></li>
<li>What we’re looking for is sequences with these 2 characteristics:
<ol type="1">
<li>labeled non-coding in the dataset (i.e.&nbsp;ground truth is non-coding)</li>
<li>consistently classified as coding across epochs.</li>
</ol></li>
</ul>
<p>The following figure demonstrates this idea: * Estimated coding probability across all training epochs shown for five example RNAs for the LSTM model. * We expect coding and non-coding RNAs to have high and low coding probabilities respectively; this is the case for examples A and E. * We are interested in lncRNAs—like B (ENST00000447563)—which have consistently high estimated coding probability, despite having the ground truth-label ‘Non-coding’. * C and D show examples of ambiguous samples, i.e.&nbsp;they show a large change in estimated coding probabilities as model training progresses, so we are not sure whether they are mislabeled or not.</p>
<p><img src="images/image1.png" alt="drawing" width="400"></p>
<p>Here’s a training dynamics summary overview plot showing what this looks like for the complete dataset:</p>
<ul>
<li>Each blue dot is an RNA with ground truth label coding. The orange dots are RNAs with ground truth label non-coding.</li>
<li>y-axis is the average of ground-truth label predictions across all epochs.</li>
<li>x-axis is the standard deviation of the ground-truth label predictions across all epochs.</li>
<li>lncRNAs in the <em>hard-to-learn</em> region of the following figure are considered candidate misannotated lncRNAs.
<ul>
<li>Why?– These samples have a low mean and standard deviation for the predicted probability of the ground-truth (i.e.&nbsp;non-coding) class overall training epochs.</li>
<li>In other words, RNAs that fall in this region are consistently classified into the non-ground-truth (i.e.&nbsp;coding) class.</li>
</ul></li>
</ul>
<p><img src="images/image2.png" alt="drawing" width="600"></p>
</section>
<section id="uh-does-this-training-dynamics-stuff-really-work-when-a-large-number-of-sequences-might-be-mislabelled" class="level3">
<h3 class="anchored" data-anchor-id="uh-does-this-training-dynamics-stuff-really-work-when-a-large-number-of-sequences-might-be-mislabelled">Uh, does this training dynamics stuff really work when a large number of sequences might be mislabelled?</h3>
<p>Yes it does!</p>
<p>We designed a computational experiment where we intentionally mislabel (i.e.&nbsp;flip ground-truth labels of coding to non-coding and vice versa) a part of the dataset.</p>
<p>After flipping the labels, we retrain the model and examine the training dynamics again.</p>
<p>Here’s what it looks like:</p>
<ul>
<li>We sampled 5% of the data with mean (ground truth) &gt;= 0.8 and std (ground truth) &lt;= 0.2.</li>
<li>These are samples for which we have medium-high confidence that the ground truth labels are correct (before).</li>
<li>After we flip the labels (coding RNAs become non-coding RNAs and vice-versa), the samples move into the <em>hard-to-classify</em> region of the training dynamics summary plot.</li>
</ul>
<p><img src="images/image3.png" alt="drawing" width="400"></p>
</section>
<section id="ok-so-this-computational-method-works.-but-how-well-do-the-results-agree-with-exisiting-experimental-results" class="level3">
<h3 class="anchored" data-anchor-id="ok-so-this-computational-method-works.-but-how-well-do-the-results-agree-with-exisiting-experimental-results">Ok, so this computational method works. But how well do the results agree with exisiting experimental results?</h3>
<p>We complied data from 4 experimental sources and found a statistically significant overlap in all cases: 1. cncRNAdb—a manually curated resource of experimentally verified coding ncRNAs 2. Ribo-Seq data-based methods used to find misannotated lncRNAs: 1. FLOSS 2. ORFScore 3. PhyloP</p>
<p><img src="images/image4.png" alt="drawing" width="200"></p>
<p><img src="images/image5.png" alt="drawing" width="800"></p>
</section>
<section id="hmm-this-is-useful-whats-the-best-way-to-use-this-data" class="level3">
<h3 class="anchored" data-anchor-id="hmm-this-is-useful-whats-the-best-way-to-use-this-data">Hmm, this is useful! What’s the best way to use this data?</h3>
<p>The best way to find a <em>really</em> high-confidence candidate misannotated lncRNA (e.g.&nbsp;for experimental validation) is to aggregate data from many sources!</p>
<p>To demonstrate how, we compiled 6 other types of data for a few misannotated lncRNAs we identified: 1. Ribo-Seq identifies RNAs associated with ribosomes, which are likely to be translated. <em>Ribo-Seq P-values</em> (a combined score from FLOSS, ORFScore and PhyloP) show the likelihood of the identified RNA being a true-positive. 2. <em>PeptideShaker</em> analyzes publicly available mass-spectrometry data and provides a confidence score for each peptide. 3. The <em>Top BLAST Hit</em> is the top hit from running the query in BLASTx. 4. The <em>Top hmmer Hit</em> is the hmmer hit with the most significant E-value obtained by running hmmscan on the Pfam profile database that looks for known domains. 5. <em>CNIT, CPC2, CPAT and RNASamba</em> are tools for the coding potential prediction of a given RNA.</p>
<p><img src="images/image6.png" alt="drawing" width="800"></p>
<ol start="6" type="1">
<li><em>Alphafold2</em> provides folding predictions and confidence for given ORFs.</li>
</ol>
<p><img src="images/image7.png" alt="drawing" width="800"></p>
</section>
<section id="and-thats-a-wrap" class="level3">
<h3 class="anchored" data-anchor-id="and-thats-a-wrap">And that’s a wrap!</h3>
<p>For more nitty-gritty details, checkout the <a href="https://academic.oup.com/bioinformatics/article/39/1/btac821/6960922">Bioinformatics</a> paper!</p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>